mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/EntityHelperUpdateMethodGenerator.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.mapper.processor.entity;
+
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.querybuilder.QueryBuilder;
+import com.datastax.oss.driver.api.querybuilder.update.UpdateStart;
+import com.datastax.oss.driver.api.querybuilder.update.UpdateWithAssignments;
+import com.datastax.oss.driver.internal.mapper.processor.MethodGenerator;
+import com.datastax.oss.driver.internal.mapper.processor.ProcessorContext;
+import com.squareup.javapoet.MethodSpec;
+import java.util.Optional;
+import javax.lang.model.element.Modifier;
+
+public class EntityHelperUpdateMethodGenerator implements MethodGenerator {
+
+  private final EntityDefinition entityDefinition;
+
+  public EntityHelperUpdateMethodGenerator(
+      EntityDefinition entityDefinition,
+      EntityHelperGenerator enclosingClass,
+      ProcessorContext context) {
+    this.entityDefinition = entityDefinition;
+  }
+
+  @Override
+  public Optional<MethodSpec> generate() {
+    MethodSpec.Builder insertBuilder =
+        MethodSpec.methodBuilder("update")
+            .addAnnotation(Override.class)
+            .addModifiers(Modifier.PUBLIC)
+            .returns(UpdateWithAssignments.class)
+            .addStatement("$T keyspaceId = context.getKeyspaceId()", CqlIdentifier.class)
+            .addStatement("$T tableId = context.getTableId()", CqlIdentifier.class)
+            .beginControlFlow("if (tableId == null)")
+            .addStatement("tableId = DEFAULT_TABLE_ID")
+            .endControlFlow()
+            .addStatement(
+                "$1T update = (keyspaceId == null)\n"
+                    + "? $2T.update(tableId)\n"
+                    + ": $2T.update(keyspaceId, tableId)",
+                UpdateStart.class,
+                QueryBuilder.class)
+            .addCode("$[return update");
```

Comment:
One of the things that are different comparing` @Insert` generator is the moment of adding `USING TIMESTAMP | USING TTL"`. According to documentation https://docs.datastax.com/en/cql/3.3/cql/cql_reference/cqlUpdate.html and in our `Update` builder the `usingTimestamp` and `usingTtl` are available on the `UpdateStart` class.
Because of that, we should do it on this line. 
Unfortunately, we don't have annotation values (`customUsingClause`) in this class. Other solution will be to move this block (https://github.com/datastax/java-driver/pull/1235/files#diff-75706beabe9b72dbf270c073d07d201fR59 to line 64)
to 
`generatePrepareRequest` method
https://github.com/datastax/java-driver/pull/1235/files#diff-25ba2cdaf8c214d0d4a6f05f1c7922acR147.
In that case, this `generate()` method will return `UpdateStart`, not the `UpdateWithAssignments` (so we will be able to append usingTtl or using TImestamp in the `generatePrepareRequest(`.
But the problem is that we will need to pass `EntityDefinition entityDefinition` to `generatePrepareRequest` metho as well. It could be done in the `EntityHelperGenerator` method: `methodGenerator.generate(entityDefinition)`
but it is not present in `DaoImplementationGenerator` nor `MapperImplementationGenerator` classes - so it will need bigger refactor. @olim7t let me know wdyt

Reply:
I see. We have two options:
1. make `EntityHelper.update` not return the statement directly, but take an existing statement and modify it. Then the DAO method would be responsible for starting the statement.
2. modify the query builder to allow `usingTimestamp/usingTtl` at later steps.
3. cast to `DefaultUpdate`, which gives us access to the methods again. (this is safe here because we control the creation of the statement, we know it's the default implementation)

I would have liked 2, but it breaks the API. So I think we should do 3.

Reply:
I was thinking about the 2nd approach but you are right - it breaks the API.
3 seems to be a clever trick to alleviate that problem. I've implemented it in this commit:
https://github.com/datastax/java-driver/pull/1235/commits/eb1c29f2338b4341a9759ca9fa9a641ea9936d62